libs/test/SalesforceSDKTest/src/com/salesforce/androidsdk/security/KeyStoreWrapperTest.java

```diff
@@ -139,4 +154,25 @@ public void testDecryptDataEncryptedWithLegacyRSACipher() {
         final String decryptedData = Encryptor.decryptWithRSA(privateKey, encryptedData);
         Assert.assertEquals("Decrypted data should match original data", data, decryptedData);
     }
+
+    /**
+     * Upgrade test: client upgraded and server upgraded
+     * 1. Client generated key pair before upgrading to 11.1.1
+     * 2. Server is upgraded and encrypts push notification with stored key using new RSA cipher mode
+     * 3. Client tries to decrypt push notification
+     */
+    @Test
+    public void testDecryptDataEncryptedWithNewCipherForKeyCreatedBeforeChange() {
```

Comment:
That scenario fails. Because when generating a key, you have to specify what encryption paddings / digest etc can be used with a given key. So a key generated pre 11.1.1 won't be allowed to decrypt using a different cipher mode even though the server will probably successfully encrypt with it using a different cipher mode.